File: python/leetcode83.py
#!/usr/bin/env python3
#coding:utf-8


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
    	if head == None:
    		return None

    	p = head;
    	while p.next != None:
    		if p.val == p.next.val:
    			p.next = p.next.next;
    		else:
    			p = p.next;

    	return head

File: cpp/src/leetcode83.cpp
//
// Created by bob on 19-3-8.
//

#include "../include/header.h"


ListNode* Solution::deleteDuplicates(ListNode* head){
    if (head == NULL)
        return head;
    ListNode* p = head;
    while (p->next != NULL){
        ListNode* tmp = p->next;
        if (p->val == tmp->val){
            p->next = tmp->next;
        } else{
            p = p->next;
        }
    }
    return head;
}